aws/rust-runtime/aws-config/additional-ci

```diff
@@ -14,11 +14,8 @@ cargo "+${RUST_NIGHTLY_VERSION:-nightly}" api-linter --all-features --config api
 echo "### Checking for duplicate dependency versions in the normal dependency graph with all features enabled"
 cargo tree -d --edges normal --all-features
 
-echo "### Testing with all features enabled"
-cargo test --all-features
```

Comment:
This one needs to say because `aws-config` is not part of the `aws/rust-runtime` workspace, so its tests are not being run by `check-rust-runtimes-and-tools` by default. Definitely deserves a comment explaining that though so we don't accidentally remove it in the future.